src/power_grid_model_io/converters/pandapower_converter.py

```diff
@@ -0,0 +1,361 @@
+        pgm_3wtransformers["u1"] = pp_trafo3w["vn_hv_kv"] * 1e3
+        pgm_3wtransformers["u2"] = pp_trafo3w["vn_mv_kv"] * 1e3
+        pgm_3wtransformers["u3"] = pp_trafo3w["vn_lv_kv"] * 1e3
+        pgm_3wtransformers["sn_1"] = pp_trafo3w["sn_hv_mva"] * 1e6
+        pgm_3wtransformers["sn_2"] = pp_trafo3w["sn_mv_mva"] * 1e6
+        pgm_3wtransformers["sn_3"] = pp_trafo3w["sn_lv_mva"] * 1e6
+        pgm_3wtransformers["uk_12"] = pp_trafo3w["vk_hv_percent"] * 1e-2
+        pgm_3wtransformers["uk_13"] = pp_trafo3w["vk_lv_percent"] * 1e-2
+        pgm_3wtransformers["uk_23"] = pp_trafo3w["vk_mv_percent"] * 1e-2
+
+        pgm_3wtransformers["pk_12"] = (pp_trafo3w["vkr_hv_percent"] * 1e-2) * (
+            pp_trafo3w[["sn_hv_mva", "sn_mv_mva"]].min(axis=1) * 1e6
+        )
+
+        pgm_3wtransformers["pk_13"] = (pp_trafo3w["vkr_lv_percent"] * 1e-2) * (
+            pp_trafo3w[["sn_hv_mva", "sn_lv_mva"]].min(axis=1) * 1e6
+        )
+
+        pgm_3wtransformers["pk_23"] = (pp_trafo3w["vkr_mv_percent"] * 1e-2) * (
+            pp_trafo3w[["sn_mv_mva", "sn_lv_mva"]].min(axis=1) * 1e6
+        )
+
+        pgm_3wtransformers["i0"] = pp_trafo3w["i0_percent"] * 1e-2
+        pgm_3wtransformers["p0"] = pp_trafo3w["pfe_kw"] * 1e3
+        pgm_3wtransformers["winding_1"] = pp_trafo3w["vector_group"].apply(get_3wdgtransformer_winding_1)
+        pgm_3wtransformers["winding_2"] = pp_trafo3w["vector_group"].apply(get_3wdgtransformer_winding_2)
+        pgm_3wtransformers["winding_3"] = pp_trafo3w["vector_group"].apply(get_3wdgtransformer_winding_3)
+        pgm_3wtransformers["clock_12"] = (round(pp_trafo3w["shift_mv_degree"] / 30.0) % 12).astype(np.int8)
+        pgm_3wtransformers["clock_13"] = (round(pp_trafo3w["shift_lv_degree"] / 30.0) % 12).astype(np.int8)
+        pgm_3wtransformers["tap_pos"] = pp_trafo3w["tap_pos"]
+        pgm_3wtransformers["tap_side"] = self._get_3wtransformer_tap_side(pp_trafo3w["tap_side"])
+        pgm_3wtransformers["tap_min"] = pp_trafo3w["tap_min"]
+        pgm_3wtransformers["tap_max"] = pp_trafo3w["tap_max"]
+        pgm_3wtransformers["tap_nom"] = pp_trafo3w["tap_neutral"]
+        pgm_3wtransformers["tap_size"] = self._get_3wtransformer_tap_size(pp_trafo3w)
+
+        self.pgm_data["three_winding_transformer"] = pgm_3wtransformers
+
+    def _generate_ids(self, key: str, pp_idx: pd.Index) -> np.arange:
+        assert key not in self.idx_lookup
+        n_objects = len(pp_idx)
+        pgm_idx = np.arange(start=self.next_idx, stop=self.next_idx + n_objects, dtype=np.int32)
+        self.idx[key] = pd.Series(pgm_idx, index=pp_idx)
+        self.idx_lookup[key] = pd.Series(pp_idx, index=pgm_idx)
+        self.next_idx += n_objects
+        return pgm_idx
+
+    def _get_ids(self, key: str, pp_idx: pd.Series) -> pd.Series:
+        if key not in self.idx:
+            raise KeyError(f"No indexes have been created for '{key}'!")
+        return self.idx[key][pp_idx]
+
+    @staticmethod
```

Comment:
Just curious: Why use static method here?

Apply function processes a single row. And maybe its possible for this function to be in vectorized form.